src/main/java/taxi/dao/jdbcimpl/DriverDaoJdbcImpl.java

```diff
@@ -0,0 +1,117 @@
+package taxi.dao.jdbcimpl;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import taxi.dao.DriverDao;
+import taxi.exception.DataProcessingException;
+import taxi.lib.Dao;
+import taxi.model.Driver;
+import taxi.util.ConnectionUtil;
+
+@Dao
+public class DriverDaoJdbcImpl implements DriverDao {
+    @Override
+    public Driver create(Driver driver) {
+        String insertQuery = "INSERT INTO drivers (driver_name, license_number) VALUES (?, ?);";
+
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement insertStatement = connection.prepareStatement(insertQuery,
+                         Statement.RETURN_GENERATED_KEYS)) {
+            insertStatement.setString(1,driver.getName());
+            insertStatement.setString(2,driver.getLicenseNumber());
+            insertStatement.executeUpdate();
+            ResultSet resultSet = insertStatement.getGeneratedKeys();
+            if (resultSet.next()) {
+                driver.setId(resultSet.getObject("GENERATED_KEY", Long.class));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can not add new driver: "
+                    + driver.toString(), e);
+        }
+        return driver;
+    }
+
+    @Override
+    public Optional<Driver> get(Long id) {
+        String selectQuery = "SELECT * FROM drivers WHERE deleted = FALSE AND drivers.id = ?;";
+
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement preparedStatement = connection.prepareStatement(selectQuery)) {
+            ResultSet resultSet = preparedStatement.executeQuery();
+            if (resultSet.next()) {
+                return Optional.of(createDriver(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can not get a driver with id: " + id, e);
+        }
+        return Optional.empty();
+    }
+
+    private Driver createDriver(ResultSet resultSet) {
```

Comment:
move this method to the bottom of the file

Reply:
Done